olp-cpp-sdk-dataservice-read/src/repositories/PrefetchTilesRepository.cpp

```diff
@@ -169,6 +191,32 @@ SubQuadsResponse PrefetchTilesRepository::GetSubQuads(
   OLP_SDK_LOG_TRACE_F(kLogTag, "GetSubQuads(%s, %" PRId64 ", %" PRId32 ")",
                       tile.ToHereTile().c_str(), version, depth);
 
+  repository::PartitionsCacheRepository repository(
+      catalog, settings.cache, settings.default_cache_expiration);
+
+  auto get_sub_quads = [](const QuadTreeIndex& tree) -> SubQuadsResult {
+    SubQuadsResult result;
+    auto index_data = tree.GetIndexData();
+    std::transform(
+        index_data.begin(), index_data.end(),
+        std::inserter(result, result.end()),
+        [](const QuadTreeIndex::IndexData& data) -> SubQuadsResult::value_type {
+          return {data.tile_key, data.data_handle};
+        });
+    return result;
+  };
+
+  // check if quad tree with requested tile and depth already in cache
+  auto cached_tree = repository.Get(layer_id, tile, depth, version);
```

Comment:
Don't we need to search for any quad tree with metadata of the tile?
Its fine for tiles from one Prefetch request, since their level differs on `kMaxQuadTreeIndexDepth+1`, but for tiles from different Prefetch requests metadata from quad tree will duplicate. We are doing search in GetData/GetAggregatedData  to avoid such extra quad trees in the cache.

Reply:
This is tricky, if we will search quad tree on different levels, we should check if it covers full tree, if not, request missing tiles. And even than we can have duplicates, because for example the missing coverage only one level in between of cached ones.